test/Microsoft.ML.Tests/OnnxTests.cs

```diff
@@ -268,6 +282,57 @@ public void KeyToVectorWithBagTest()
             Done();
         }
 
+        [Fact]
+        public void KmeansTest()
+        {
+            string dataPath = GetDataPath(@"breast-cancer.txt");
+            var pipeline = new Legacy.LearningPipeline(0);
+
+            pipeline.Add(new Legacy.Data.TextLoader(dataPath)
+            {
+                Arguments = new TextLoaderArguments
+                {
+                    Separator = new[] { '\t' },
+                    HasHeader = true,
+                    Column = new[]
+                    {
+                        new TextLoaderColumn()
+                        {
+                            Name = "Features",
+                            Source = new [] { new TextLoaderRange(1, 9) },
+                            Type = Legacy.Data.DataKind.R4
+                        },
+                    }
+                }
+            });
+
+            pipeline.Add(new KMeansPlusPlusClusterer() { K = 2, MaxIterations = 1, NumThreads = 1, InitAlgorithm = KMeansPlusPlusTrainerInitAlgorithm.Random });
+            var model = pipeline.Train<BreastNumericalColumns, BreastCancerClusterPrediction>();
+            var subDir = Path.Combine("..", "..", "BaselineOutput", "Common", "Onnx", "Cluster", "BreastCancer");
+            var onnxPath = GetOutputPath(subDir, "Kmeans.onnx");
+            DeleteOutputPath(onnxPath);
+
+            var onnxAsJsonPath = GetOutputPath(subDir, "Kmeans.json");
+            DeleteOutputPath(onnxAsJsonPath);
+
+            OnnxConverter converter = new OnnxConverter()
+            {
+                Onnx = onnxPath,
+                Json = onnxAsJsonPath,
+                Domain = "Onnx"
+            };
+
+            converter.Convert(model);
+
+            // Strip the version.
+            var fileText = File.ReadAllText(onnxAsJsonPath);
+            fileText = Regex.Replace(fileText, "\"producerVersion\": \"([^\"]+)\"", "\"producerVersion\": \"##VERSION##\"");
+            File.WriteAllText(onnxAsJsonPath, fileText);
+
+            CheckEquality(subDir, "Kmeans.json");
```

Comment:
> CheckEquality(subDir, "Kmeans.json"); [](start = 11, length = 38)

Is checking for hard equality like this something that we usually do for ONNX tests? If something like the opset version changes later to a newer ONNX, this will requiring fixing the test. Are we ok with that?

Reply:
Yes, all existings are based on model file comparison, so we of course need to fix that. In the future, we should have a new test framework for running real tests.